src/hardware/rdtech-um/api.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of the libsigrok project.
+ *
+ * Copyright (C) 2018-2020 Andreas Sandberg <andreas@sandberg.pp.se>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <config.h>
+#include <glib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include <libsigrok/libsigrok.h>
+#include "libsigrok-internal.h"
+
+#include "protocol.h"
+
+#define RDTECH_UM_SERIALCOMM "115200/8n1"
+
+static const uint32_t scanopts[] = {
+	SR_CONF_CONN,
+	SR_CONF_SERIALCOMM,
+};
+
+static const uint32_t drvopts[] = {
+	SR_CONF_MULTIMETER,
```

Comment:
There is a config key (device type) `SR_CONF_ENERGYMETER` which is perhaps a better fit for this device?

Reply:
Would it be appropriate to list it as both `SR_CONF_MULTIMETER` and `SR_CONF_ENERGYMETER`?  I'm not entirely sure how this is used by the library users.

Reply:
There are devices that have more than one device type (like scopes with digital channels), but technically speaking this isn't a multimeter with switchable measurement types.

Only place I'm aware of where the device type is evaluated, is in SmuView. :)
It is f.e. used to show different GUI elements for different device types and to auto generate math channels for power supplies and loads.
Though at the moment, multimeters and power meters are both handled as "measurement devices". So no special treatment for power meters, right now.

Reply:
So maybe it should be listed as a combination of `SR_CONF_POWERMETER` and `SR_CONF_ENERGYMETER` in that case? I'm honestly not sure why power meters and energy meters have their own types, it seems like most energy meters would also measure power and as far as I can tell, there is exactly one device of either type in Sigrok at the moment (both are listed as energy meters on the wiki though).